site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,627 @@
+a Java object automatically. So *Armeria* supports that using `@RequestConverter`_ and `@RequestObject`_
+annotations. You can write your own request converter by implementing RequestConverterFunction_ as follows.
+Similar to the exception handler, you can call ``RequestConverterFunction.fallthrough()`` when your request
+converter is not able to convert the request.
+
+.. code-block:: java
+
+    class MyRequestConverter implements RequestConverterFunction {
+        @Override
+        public Object convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage request,
+                                     Class<?> expectedResultType) throws Exception {
+            if (expectedResultType == MyObject.class) {
+                // Convert the request to a Java object.
+                return new MyObject(request.content());
+            }
+
+            // To the next request converter.
+            return RequestConverterFunction.fallthrough();
+        }
+    }
+
+Then, you can write your service method as follows. Note that a request converter will work on the parameters
+which are annotated with `@RequestObject`_.
+
+.. code-block:: java
+
+    @RequestConverter(MyRequestConverter.class)
+    class MyAnnotatedServiceClass {
+
+        @Post("/hello")
+        public HttpResponse hello(@RequestObject MyObject myObject) {
+            // MyRequestConverter will be used to convert a request.
+            ...
+        }
+
+        @Post("/hola")
+        @RequestConverter(MySpanishRequestConverter.class)
+        public HttpResponse hola(@RequestObject MySpanishObject myObject) {
+            // MySpanishRequestConverter and MyRequestConverter will be used to convert a request.
+            ...
+        }
+    }
+
+*Armeria* also provides built-in request converters such as, JacksonRequestConverterFunction_ for JSON
+documents, StringRequestConverterFunction_ for text contents and ByteArrayRequestConverterFunction_
+for binary contents. They will be applied after your request converters by default, so you can use these
+built-in converters by just putting `@RequestObject`_ annotation on the parameters which you want to convert.
+
+In some cases, a parameter of a method may be annotated with `@RequestConverter`_. Assume that you have
+a Java class named ``MyRequest`` that it is usually able to be converted by ``MyDefaultRequestConverter``.
+But what if there is only one method which has a parameter of ``MyRequest`` that you have to convert
+it differently. In this case, you may annotate a variable with `@RequestConverter`_ as follows.
+In the example, ``MySpecialRequestConverter`` will be used first for converting ``MyRequest``.
+
+.. code-block:: java
+
+    @RequestConverter(MyDefaultRequestConverter.class)
+    class MyAnnotatedServiceClass {
+
+        @Post("/hello")
+        public HttpResponse hello(
+            @RequestConverter(MySpecialRequestConverter.class) MyRequest myRequest) {
```

Comment:
Don't we need to annotate with `@RequestObject` as well? If not, we could mention that a user doesn't have to annotate with `@RequestObject` when a parameter is annotated with `@RequestConverter`.

Reply:
Thanks. It was my mistake, it had to be `@RequestObject`, not `@RequestConverter`. I fixed it. :-)